csup/dict.go

```diff
@@ -2,113 +2,71 @@ package csup
 
 import (
 	"io"
-	"math"
 
 	"github.com/brimdata/super"
-	"github.com/brimdata/super/zcode"
 	"golang.org/x/sync/errgroup"
 )
 
 type DictEncoder struct {
-	typ    super.Type
-	tags   map[string]byte
-	index  []byte
+	typ super.Type
+	PrimitiveEncoder
```

Comment:
Nit: We usually put embedded fields first.
```suggestion
	PrimitiveEncoder
	typ super.Type
```